Persistence/Repositories/Impl/QuestionRepository.cs

```diff
@@ -11,45 +12,79 @@ public QuestionRepository(AppDbContext context)
         _context = context;
     }
 
-    public ICollection<Question> GetQuestions()
+    public async Task<ICollection<Question>> GetQuestions()
     {
-        return _context.Questions!.OrderBy(q => q.Id).ToList();
+        return await _context.Questions!
+            .Include(q => q.Author)
+            .Include(q => q.QuestionTags)
+            .Include(q => q.Answers)
+            .OrderBy(q => q.Id)
+            .ToListAsync();
     }
 
-    public Question? GetQuestion(long id)
+    public async Task<Question?> GetQuestion(long id)
     {
-        return _context.Questions!.FirstOrDefault(q => q.Id == id);
+        var question = await _context.Questions!
+            .Include(q => q.Author)
+            .Include(q => q.QuestionTags)
+            .Include(q => q.Answers)
+            .FirstOrDefaultAsync(q => q.Id == id);
+
+        if (question == null)
+            return null;
+
+        foreach (var questionTag in question.QuestionTags!)
+        {
+            var tag = await _context.Tags!.FirstOrDefaultAsync(e => e.Id == questionTag.TagId);
```

Comment:
Why "e" and not "t"?
```suggestion
            var tag = await _context.Tags!.FirstOrDefaultAsync(t => t.Id == questionTag.TagId);
```